### PR TITLE
feature: AntdUiProvider

### PR DESCRIPTION
**Pull request by davecoates**

### CODE REVIEW

**Comment by fanglong-alliance** [comment]
we're missing Calendar; think we should have a default support for Calendar too? (yes not using it currently, otoh might want to do it now so future uses would be safer)

File: js-packages/@prestojs/ui-antd/src/AntdUiProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        if (!this.components.DatePicker) {
+            throw new Error(`To use components that use DatePicker you must first configure @prestojs/ui-antd so it knows the type to use. To set the default moment based datepicker add the following to your app before any widgets are used:
+
+    import { DatePicker } from 'antd';
+
+    <AntdUiProvider datePickerComponent={DatePicker}>...</AntdUiProvider>
+
+See https://prestojs.com/docs/ui-antd/ for instructions on setting up @prestojs/ui-antd and 
+https://ant.design/docs/react/replace-moment#DatePicker for options in customising the DatePicker
+`);
```
Comment (fanglong-alliance) [inline js-packages/@prestojs/ui-antd/src/AntdUiProvider.tsx:64]: i think we should mention that moment is now legacy and generally discouraged
(ant's not doing that for major version backwards compatibility reason but we dont have that burden)

**Comment by fanglong-alliance** [comment]
just a note, would need to change rangefields once they're in as well. 
lgtm

**Comment by davecoates** [comment]
> just a note, would need to change rangefields once they're in as well.
> lgtm

ah yep - forgot to mention that (didn't want to introduce conflicts so intentionally left it out)

**Comment by davecoates** [comment]
> we're missing Calendar; think we should have a default support for Calendar too? (yes not using it currently, otoh might want to do it now so future uses would be safer)

I'm not sure there's any benefit doing it now - may even mislead into thinking it's available somewhere. Would also mean project pulls in Calendar even though never used anywhere (we'd use lazy load but still generates another bundle for no reason).

**Pull request merged**